### ISSUE TITLE
quarkus does not support automatic module in gradle

### ISSUE THREAD

**Issue by arjenzhou** [quarkusio/quarkus#19560]
### Describe the bug

**Environment:**
- macOS 11.5.2
- Java 11
- Gradle 7.2
- Intellij IDEA 2021.2 CE

**Behavior**
In `module-info.java`, requires quarkus but get `module not found` after building. 

For any automatic module, there should be a `Automatic-Module-Name` in `MANIFEST.MF` or,
adding a `module-info.java` in source root.

**Comment by gastaldi** [comment]
Can you elaborate on what you're trying to do here? Quarkus does not support JPMS, so you can't use Quarkus as a module anyway

**Comment by arjenzhou** [comment]
> Can you elaborate on what you're trying to do here? Quarkus does not support JPMS, so you can't use Quarkus as a module anyway

Sad to hear that. I am writing code under JPMS, is there any plan to support it?

**Comment by gastaldi** [comment]
Not really, how are you using Quarkus? You shouldn't need JPMS while writing a Quarkus app

**Comment by arjenzhou** [comment]
> Not really, how are you using Quarkus? You shouldn't need JPMS while writing a Quarkus app

I `requires` Quarkus (quarkus.core etc.) in `module-info.java`, which is necessary for importing Quarkus lib. Is there any inaccuracy？

**Comment by gastaldi** [comment]
That's not how you are supposed to develop with Quarkus. Take a look at the getting started example: https://github.com/quarkusio/quarkus-quickstarts/tree/main/getting-started and the guide in https://quarkus.io/get-started/

**Comment by arjenzhou** [comment]
> That's not how you are supposed to develop with Quarkus. Take a look at the getting started example: https://github.com/quarkusio/quarkus-quickstarts/tree/main/getting-started and the guide in https://quarkus.io/get-started/

Alright, I had checked the example. It seems hard to migrate a JPMS app to Quarkus. Writing an unnamed module is something strange to me.
